fw/ss_skb.h

```diff
@@ -211,6 +216,44 @@ ss_skb_adjust_data_len(struct sk_buff *skb, int delta)
 	skb->truesize += delta;
 }
 
+/**
+ * Insert a single sk_buff or list of SKBs
+ * pointed to by @nskb_head after the specified @skb.
+ */
+static inline void
+ss_skb_queue_insert_after(struct sk_buff *skb, struct sk_buff *nskb_head)
+{
+	struct sk_buff *next = skb->next;
+
+	next->prev = nskb_head->prev;
+	nskb_head->prev->next = next;
+
+	nskb_head->prev = skb;
+	skb->next = nskb_head;
+}
+
+#define ss_skb_queue_for_each_do(queue, lambda)		\
+do {							\
+	int i = 0;					\
+	struct sk_buff *skb = *queue;			\
+	if (likely(skb)) {				\
+		do {					\
+			lambda;				\
+			skb = skb->next;		\
+		} while (skb != *queue);		\
+	}						\
+} while(0)
```

Comment:
Should also be under `DEBUG` macro - in this way a reader won't be nervous about an skb list traversal hidden somewhere in the code.